src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -509,6 +514,10 @@ internal override void AfterAddingTypeMembersChecks(ConversionsBase conversions,
                 {
                     diagnostics.Add(ErrorCode.ERR_BadVisIndexerParam, Location, this, param.Type);
                 }
+                else if (param.Type.IsFileTypeOrUsesFileTypes() && !this.ContainingType.IsFileTypeOrUsesFileTypes())
```

Comment:
Should we care about `IsExplicitInterfaceImplementation` like we do in case above?

Reply:
Let's make it an open question to LDM. I'd prefer to keep the rules simple for now--"if you're a file type, you can use file types in your member signatures".

My intuition is that it's fine to explicitly implement an interface member that uses `file` types even when we aren't a `file` type. Particularly since a non-`file` type can implement a `file interface`. Just something to bring along later.

Reply:
How are you tracking this?

Reply:
Added a line to the test plan #60819.